### PR TITLE
Swapna_Assignment_24_Suggested_Changes_Done_30_Min

### PR DESCRIPTION
**Pull request by SwapnaWJun20**
image

### CODE REVIEW

File: src/swapnaWJun20/MobileNoValidity.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		}
+		return flag;
+	}
+
+	void checkMobileNo(String mobNo) {
+		if (mobNo.length() > 14 || mobNo.length() < 10) {
+			System.out.println("\n" + mobNo + " : Mobile no is invalid.");
+		} else {
+			if (mobNo.length() == 10 && mobNo.startsWith("9") || mobNo.startsWith("8") || mobNo.startsWith("7")) {
+				checkAlphabet(mobNo);
```
Comment (AnupSJun20) [inline src/swapnaWJun20/MobileNoValidity.java]: Here, in the If condition, we can use directly the method calling as you are returning the value and using that. One checkAlphabet(mobNo) calling and if(flag == true) not required together.

File: src/swapnaWJun20/MobileNoValidity.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		mobileNoValidity.checkMobileNo("09765463742");
+		mobileNoValidity.checkMobileNo("976 546 3742");
+		mobileNoValidity.checkMobileNo("9765463742");
+
+		mobileNoValidity.checkMobileNo("998979765463742"); // invalid
+		mobileNoValidity.checkMobileNo("0765463742");// invalid
+		mobileNoValidity.checkMobileNo("99897  83742"); // invalid
+		mobileNoValidity.checkMobileNo("998 97 83742");// invalid
+		mobileNoValidity.checkMobileNo("998 979 S7Y2");// invalid
+	}
```
Comment (AnupSJun20) [inline src/swapnaWJun20/MobileNoValidity.java:63]: Please enter a number with 979 567894, although it is 9 digits and with a space it becomes 10 as the length, so the code returns this as a valid number. Please add the space criteria.
Rest you have covered all the conditions well.

**Review by AnupSJun20 — CHANGES_REQUESTED** [review]
Please incorporate the changes. Covered almost all scenarios.

File: src/swapnaWJun20/MobileNoValidity.java (hunk 2 above)
Comment (SwapnaWJun20) [reply src/swapnaWJun20/MobileNoValidity.java:63]: I have added the check specified above to the code.

File: src/swapnaWJun20/MobileNoValidity.java (hunk 1 above)
Comment (SwapnaWJun20) [reply src/swapnaWJun20/MobileNoValidity.java]: Updated the code as per the comments.

**Review by AnupSJun20 — APPROVED** [review]
Thanks for adding the validation. Code looks good. Approving.

**Pull request merged**